packages/block-editor/src/components/block-alignment-toolbar/index.js

```diff
@@ -89,10 +89,10 @@ export function BlockAlignmentToolbar( {
 
 export default compose(
 	withSelect( ( select ) => {
-		const { getSettings } = select( 'core/block-editor' );
-		const settings = getSettings();
+		const { getThemeSupports } = select( 'core' );
```

Comment:
The block editor package shouldn't be relying on `core-data` it's a network/persistence agnostic package that allows creating third-party editors. It means, this should still use getSettings.

Reply:
Just to clarify, are you saying that `@wordpress/block-editor` should _never_ fetch any information from the network? In conclusion, that would mean that it'll _always_ depend on data passed through a global variable -- i.e. it'll be always tightly coupled to server-side provided information.

But wouldn't that mean it'd be impossible to ever implement a client-side only editor?

Reply:
> But wouldn't that mean it'd be impossible to ever implement a client-side only editor?
 
For me it's the opposite, it doesn't mean that it's easier to implement client-side only editors since it doesn't need to fetch anything.

Reply:
So you would have to provide the settings object yourself via whatever method during load. That could be via PHP like in Gutenberg, but could be via your own API and init code

Reply:
The current settings are something of a loosely related set of properties from different sources:
- theme options (wide alignment, custom fonts/colors, post formats support, ...)
- image sizes (commonly affected by the theme, too)
- registered block patterns and categories
- information about the edited post's lock state
- some user options
- some site options
- customized labels for certain UI elements (`write_your_story`, `enter_title_here`)

That alone feels suboptimal and even third-party editor authors would probably appreciate the settings being split into multiple interfaces.

For an SPA-style browser client that talks to the server only through REST API, this means waiting for some 5+ REST requests to finish before even starting to initialize the editor.

Many of them are not even used by the `block-editor` package, like `imageSizes` or `imageDimensions`. These could be fetched just-in-time by the blocks that need them.

Is there agreement that the settings split described above is desirable? In other words, is the objection solely about the `alignWide` flag (directly used by the `block-editor` package)?

Reply:
We should hold on doing any changes here because right now we're doing some changes to how these settings get the editor. Basically the source of truth for these will be theme.json in WordPress, they'll arrive to the block-editor in a single properly called "__experimentalFeatures" right now (subject to change). See this issue for more details https://github.com/WordPress/gutenberg/issues/20588

Once we finish the migration there for all the properties, we'll be able to decide the best naming...

Reply:
Other things like "patterns", "lock state", "upload handler"... are different though and I believe we don't have a good answer for these at the moment. 

 - Patterns can become registration based like block types but I'm not sure we should commit to this for now, I'd like to give patterns more time to mature before making any formal JS APIs (they're experimental now)
 - the other configs that are not theme related are more "historical". I believe most of theme still make sense as block editor settings as their meaning can be different from one editor to another but If there's concrete suggestions to change one of these we can discuss it.

That said, I believe it's a hard rule that the "block-editor" package shouldn't do any REST API call on its own.

Reply:
Fair enough. In that case, it'd be good if we could have a different mechanism to pass these options on the client side (after obtaining them through an external mechanism) -- maybe either as a prop, or maybe through a filter?

(Assuming that those settings are indeed invariant, and shouldn't be subject to change by the client, or at least not the block editor -- which makes sense for e.g. "theme supports" flags.)